### ISSUE TITLE
.NET Core support

### ISSUE THREAD

**Issue by LeeCampbell** [HdrHistogram/HdrHistogram.NET#17]
Ideally HdrHistogram should be available on all supported versions of .NET.
Currently this is 4.5.2 & 4.6.1
It may however be prudent to wait until .NET Core and supporting tooling is available before undertaking this task, as it may incur alot of rework if the goalposts move by too much.

**Comment by alhardy** [comment]
Any movement on a .net core or better yet a build for .net standard?

**Comment by LeeCampbell** [comment]
There was an early attempt to port to DNX, but just as @qooroo and I started making progress, things changed (beta, dnx vs dotnet, pcl/core/standard etc). Now that .NET Standard seems to be settling down as the way to package libs, I will look at that.
Help is appreciated, else I will probably see what Json.NET has done and copy him.

**Comment by alhardy** [comment]
OK, this is only my list, happy to help out when I get to it if not done by then

**Comment by alhardy** [comment]
@LeeCampbell I've ported the main library to .NET Standard, there wasn't too much to do there.

There is however an issue with the benchmarking project, there's no .net core build for `BenchmarkDotNet.Diagnostics.Windows` so we wouldn't be able to make use of the `MemoryDiagnoser`

I could leave the `HdrHistogram.Benchmarking` as a .net 4.6.1 project however csproj with xproj references don't work, the benchmark project would have to reference `HdrHistogram` as a nuget package.

Thoughts?

**Comment by alhardy** [comment]
Actually I can reference `HdrHistogram.xproj` using a reference with a hint path rather than a project reference, happy with this approach? i.e.

```
 <Reference Include="HdrHistogram">
      <HintPath>..\HdrHistogram\bin\Debug\net461\HdrHistogram.dll</HintPath>
    </Reference>
```

**Comment by LeeCampbell** [comment]
Thanks for chasing this up @alhardy. The goal of the MemoryDiagnoser in the Benchmark is purely to ensure that we are not allocating on the hot path. However, this isn't part of an automated rig that can fail the build if we do. So if we cant have the .NET Framework specific code then so be it.

However I do like your work around. Are you able to set up a Pull Request? I will pull down locally and see if it plays nicely on my machine too.

**Comment by alhardy** [comment]
@LeeCampbell  pushing now, I've also updated the unit test and example project to .net core.

We'll still have to look at the build script though, most of it can be retired as the dotnet cli will take care of of packaging, snk etc. Would you be able to take care of this?

**Comment by alhardy** [comment]
There's my commit ^

I won't create a PR yet as the build script won't be working. Let me know if you want me to take a look at this or if your happy to.

**Comment by alhardy** [comment]
One other thing, there doesn't seem to be an unit equivalent to `xunit.runner.visualstudio` allowing nunit tests to be run in visual studio for .net core, I'm using resharper which works fine otherwise running `dotnet test` in the unit test project works as well.

**Comment by alhardy** [comment]
@LeeCampbell I've pushed a working build script in the commit above, you'll need to install the VS Update 3 and latest dotnet cli https://www.microsoft.com/net/core#windowsvs2015

I've moved the solution file and test project for this which is following the now default project structure

**Comment by mattwarren** [comment]
> There is however an issue with the benchmarking project, there's no .net core build for `BenchmarkDotNet.Diagnostics.Windows` so we wouldn't be able to make use of the `MemoryDiagnoser`

@alhardy, this will change quite soon, see https://github.com/dotnet/BenchmarkDotNet/issues/308#issuecomment-263229643 for more info

**Comment by alhardy** [comment]
@mattwarren  were benchmarks run on every build or something that was run adhoc. For now we could add another script to run benchmarks independently and/or have flag on the build script to call out to this?

Otherwise wait for dotnet core support on benchmarkdotnet diagnostics?

**Comment by LeeCampbell** [comment]
Nice. Either a separate build script or a flag is good for me.

On 29 Nov. 2016 4:35 am, "Allan Hardy" <notifications@github.com> wrote:

@mattwarren <https://github.com/mattwarren> were benchmarks run on every
build or something that was run adhoc. For now we could add another script
to run benchmarks independently and/or have flag on the build script to
call out to this?

Otherwise wait for dotnet core support on benchmarkdotnet diagnostics?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/HdrHistogram/HdrHistogram.NET/issues/17#issuecomment-263386360>,
or mute the thread

**Comment by adamsitnik** [comment]
> Otherwise wait for dotnet core support on benchmarkdotnet diagnostics?

We are going to release v 0.10.1 of BenchmarkDotNet with cross-platform Memory Diagnoser within this week. If you don't want to wait you can simply add our dev CI feed and use it today:

`<add key="appveyor-bdn" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />`

**Comment by alhardy** [comment]
@adamsitnik Nice, I'll wait for that ;)

**Comment by alhardy** [comment]
Didnt get back to this since waiting for the benchmarkdot update. Ill give it another go end of this week

**Comment by dtillman** [comment]
@LeeCampbell : Hi, could you provide status on this. When will we be able to reference/restore a release that has .NET Core and .NET Framework support.

Thanks

**Comment by LeeCampbell** [comment]
Really sorry about the delay on this one.
Plan is to pave my machine and get Vs2017 and the new .NET Core tools installed.
Hopefully they work nicely. This will be the 4th attempt at doing this. First attempt was back in 2015 :-(

**Comment by dtillman** [comment]
@LeeCampbell : No problem... is there a nuget feed I can restore from that has builds of this version (i.e. .NET Core and .NET Framework supported) available.  Maybe the appveyor project feed?

**Comment by LeeCampbell** [comment]
@dtillman Can you check if https://ci.appveyor.com/nuget/hdrhistogram-net-7o4w2wn7lgf9 gives you what you need.
Version `HdrHistogram.1.24.0-eekpfsxs-PR52.nupkg` is the result of the #52 Pull Request. Currently untested, but it would be great to get your feedback if you have the time/appetite/motivation

**Comment by dtillman** [comment]
@LeeCampbell : Thanks.. I've pulled it in and I'm currently using it/testing and all seems to be working fine on .NET Core 1.1 ... If I run into any issues I'll be sure to report...  
Let me know if you publish a pre-release someplace .. and I'll switch over to that...
Thanks again for the support!
Dave

**Comment by LeeCampbell** [comment]
A complete re-attempt to solve this with VS2017 on a clean Win10 PC is available from #58 and via nuget on https://ci.appveyor.com/nuget/hdrhistogram-net-7o4w2wn7lgf9 as version `2.24.0-xeywwypu-PR58`
